### DISCUSSION TITLE
Congo 2.0 has arrived!

### DISCUSSION THREAD

**Discussion by jpanther** [jpanther/congo#97] [Announcements]
Congo v2.0 is here and it's packed full of improvements and changes. 🎉

Built with Tailwind CSS 3.0, it is now more powerful and responsive than ever with full support for multilingual sites. Images are now automatically resized and articles gain a bunch of new display features including tables of contents, code copy and taxonomy listings.

For a complete rundown, check out the [What's new in 2.0](https://jpanther.github.io/congo/docs/version-2/) page. The documentation also contains an [upgrade guide](https://jpanther.github.io/congo/docs/version-2/upgrade/) to help move your project from version 1.x of the theme.

I hope you enjoy this release! If you have any feedback or suggestions, be sure to [open a discussion](https://github.com/jpanther/congo/discussions).

### Added
- Multilingual support
- Right-to-left (RTL) language support
- Site search powered by Fuse.js
- Automatic Markdown image resizing and srcset generation
- Performance and Accessibility improvements to achieve perfect Lighthouse scores
- Tables of Contents on article pages
- Code copy buttons in article content
- Taxonomy and term listings now support Markdown content
- Taxonomies on article and list pages
- Article pagination direction can be inverted
- Author `headline` parameter
- Skip to content and Scroll to top links
- Archetype for generating links to external articles

### Changed
- ⚠️ Required Hugo version is now 0.87.0 or later
- ⚠️ Complete rewrite of dark mode to allow more flexibile configuration
- ⚠️ All theme images are now Hugo assets
- ⚠️ Overhauled `figure` shortcode which now resizes images
- Upgrade to Tailwind v3.0.18
- Inline Javascript moved to external files
- Improved JSON-LD structured data
- Breadcrumbs now fallback to section name when `title` is not provided
- Minor style and layout improvements

<hr /><em>This discussion was created from the release <a href='https://github.com/jpanther/congo/releases/tag/v2.0.0'>v2.0.0</a>.</em>

**Comment by jcla490** [comment]
Thank you, this looks great! One minor note, emojis of the form `:smiley_face:` do not get emojified in the article table of contents. Adding emojify to  `{{ .TableOfContents }}` in `toc.html` partial does the trick 🙏

**Reply by jpanther** [reply]
Thanks for this, I'll fix it up in the next patch.

**Reply by jpanther** [reply]
Fixed in v2.0.1

**Comment by AntoineSoetewey** [comment]
Thanks for this major update !

I followed your guide to upgrade. Everything works fine.

Just one quick question regarding the section about moving assets (https://jpanther.github.io/congo/docs/version-2/upgrade/#step-4-move-assets. At the moment, I have copied and pasted all my files from the [static folder](https://github.com/AntoineSoetewey/antoinesoetewey.com/tree/master/static) to the [asset folder](https://github.com/AntoineSoetewey/antoinesoetewey.com/tree/master/assets), so the two folder contain exactly the same files. Is it the right way to do it? or should I put them in the asset folder and delete them in the static folder? or should I do this for only some of the files, but not all?

For future files, should I put them in the static or asset folder?

This is not clear to me, sorry.

Regards,
Antoine

**Reply by jpanther** [reply]
It depends what these other files in your assets folder are. The favicons are static as well as any images or files that you are linking to directly in your content. The things you need to move are images that are used by the theme. I might amend the documentation to be a little clearer about this.

**Reply by AntoineSoetewey** [reply]
Thank you for the clarification!

**Comment by leifg** [comment]
Great update and great update guide. One thing that I missed is:

Rename `menus.toml` to `menus.en.toml` (or whatever language is your first language).

**Reply by jpanther** [reply]
Thanks for picking this up. I'll adjust the documentation to add this step.

**Comment by mesartwell** [comment]
small typo in the [git command](https://jpanther.github.io/congo/docs/version-2/upgrade/#upgrade-using-git) - s/upgrade/update/:

`git submodule update --remote --merge`

Thanks for the update! Love the search feature.

**Reply by jpanther** [reply]
Thanks, I'll fix this up!
